thirdparty/kyaml/runfn/runfn.go

```diff
@@ -146,6 +150,38 @@ func (r RunFns) getNodesAndFilters() (
 	return buff, fltrs, outputPkg, nil
 }
 
+// validateInputs checks that the inputs are KRM resources
+func (r RunFns) validateInputs(inputs *[]kio.Reader) error {
+	var result []kio.Reader
+	var nodes []*yaml.RNode
+	for i := range *inputs {
+		n, err := (*inputs)[i].Read()
+		if err != nil {
+			return fmt.Errorf("input resource list must contain only KRM resources:\n%s", err.Error())
+		}
+		// we need to write the nodes back to the buffer so that the function
+		// can read them upon execution
+		input := &bytes.Buffer{}
+		err = kio.ByteWriter{Writer: input}.Write(n)
+		if err != nil {
+			return err
+		}
+		result = append(result, []kio.Reader{&kio.ByteReader{Reader: input}}...)
+		nodes = append(nodes, n...)
+	}
+	nodes, err := (&filters.IsLocalConfig{
```

Comment:
add a comment explaining how this interacts with local config or meta resources (these are distinct concepts).